src/probes/journal.c

```diff
@@ -151,17 +161,11 @@ static bool process_existing_entries(sd_journal *journal)
         ret = read_new_entries(journal);
         if (ret < 0) {
                 return false;
-        }
-
-        if (!payload) {
+        } else if (ret == 0) {
```

Comment:
Ah if I would have read a little further on I'd have seen it.

So I'm wondering if before returning when send_data fails, we could put the data back onto the journal so it can be attempted to be read next time.

Also ret could be 0 when send_data fails so the no existing entries log message would be wrong. I'd return -1 in the send_data case and hopefully there is a way to get data back into the journal after pulling it out.